internal/fs/tarfs/tarfs.go

```diff
@@ -135,8 +136,16 @@ func (tfs *TarFS) indexEntries() error {
 		header, err := tr.Next()
 		if err != nil {
 			if errors.Is(err, io.EOF) {
+				if len(tfs.entries) == 0 {
+					// indicates the given file is empty
+					return fmt.Errorf("{%s}: %w", tfs.path, errdef.ErrInvalidTarFile)
```

Comment:
nit: Should we further call out that the file is empty?

Reply:
yes a simple `file is empty` would be clearer.